### DISCUSSION TITLE
Feeds import puts 12x12 meetings in notes

### DISCUSSION THREAD

**Discussion by anchovie91471** [code4recovery/12-step-meeting-list#1071] [General]
### Contact Details

anthony@bustedspring.com

### What happened?

First, thanks for the amazing plugin and all the hard work you have put in for the recovery community!! I used the plugin on several sites for districts & our area. It's a huge help!

I identified a bug today in the latest version:

When importing meetings from a JSON feed. The 12x12 meeting type gets imported into the Meeting Notes field instead of the meeting types array.

This appears to be due to a failing if statement on line 1521 of includes/functions.php.
`if (array_key_exists($upper_type, $upper_types))`

The key for all 12x12 meetings in the variables.php file is set to "12x12" (lowercase x) instead of "12X12" so it fails this test and gets put in the notes field.

I sent an email a week or two ago to info@code4recovery.org asking for info regarding how best to do pull requests but didn't hear back. I'd be happy to do the PR and fix if that's helpful.

**Comment by tim-rohrer** [comment]
I apologize; I missed your email from a couple weeks ago. I just searched for it, and it is there, so this is my simply having missed it. I'm glad you mentioned it. I'll provide a more detailed response there.

I am going to convert this issue to a discussion as we ask for conversations to start there, and then C4R developers use and manage the issues. As you'll see in my comment when I open the Discussion, I don't believe this is a bug. I'll explain there :-)

**Reply by tim-rohrer** [reply]
Let me elaborate. You're are correct about why `12X12` was put in the notes. The official type in the spec is `12x12`, with a lower case x.

We could change the importer to recognize a capital X, but I believe that might create problems when importing to Meeting Guide which likely expects the types to be compliant with the spec.

My suggestion with your csv files is to stick with the type as defined in the spec: `12x12`.

**Reply by joshreisner** [reply]
i think this is a legit bug actually - on our demo site we imported from orange county, and they have a few 12x12 meetings [like this one](https://code4recovery.org/meetings/the-other-12-12) which are not importing the type correctly

**Reply by anchovie91471** [reply]
Thanks for the quick reply!! Sorry about putting this under Issues. I'll make sure I put anything else here going forward.

To be clear, I am importing from a JSON feed created by the TSML plugin installed on a different site (not a CSV). The types portion of the feed looks like:

`"types":
        [
            "12x12",
            "O"
        ],`

When it's imported, the meeting looks like this in the backend, but on the originating site, the "12 Steps and 12 Traditions" checkbox is checked.

![image](https://user-images.githubusercontent.com/9162021/217995171-c35e3d84-ae68-468b-b382-945b602b86fb.png)

I definitely see why you wouldn't want to change the spec for something as minor as this. 

It feels a little hacky but you could change lines 1521 - 1522.
From
`if (array_key_exists($upper_type, $upper_types)) {`
`$meetings[$i]['types'][] = $upper_type ;`
to
`if (array_key_exists($upper_type, $upper_types)) {`
`$meetings[$i]['types'][] = $upper_type !== '12X12' ? $upper_type : $type;`

One other option would be lines 1521 - 1522
From
`if (array_key_exists($upper_type, $upper_types)) {`
`$meetings[$i]['types'][] = $upper_type;`
To
`if (array_key_exists($type, $upper_types)) {`
`$meetings[$i]['types'][] = $type;`

I don't have enough context to know why the $type is being converted to uppercase so this might not work.

**Reply by tim-rohrer** [reply]
Good to see this verified by Josh. We can discuss it tomorrow and open an issue/fix.
